x/payments/spec/03_messages.md

```diff
@@ -0,0 +1,145 @@
+| Authorised             | `bool`             | Status of authorisation
+
+
+```go
+type MsgSetPaymentContractAuthorisation struct {
+	PayerDid          did.Did `json:"payer_did" yaml:"payer_did"`
+	PaymentContractId string  `json:"payment_contract_id" yaml:"payment_contract_id"`
+	Authorised        bool    `json:"authorised" yaml:"authorised"`
+}
+```
+This message is expected to fail if:
+- any editable field violates the restrictions set for the same field in `MsgSetPaymentContractAuthorisation`
+- all editable fields are `"[do-not-modify]"`
+
+## MsgGrantDiscount 
+
+The owner of a bond can edit some of the bond's parameters using `MsgGrantDiscount`.
+
+| **Field**              | **Type**           | **Description**                                                                                               |
+|:-----------------------|:-------------------|:--------------------------------------------------------------------------------------------------------------|
+| SenderDid              | `did.Did`          | Initiator DID 
+| PaymentContractId      | `string`           | ID for the paymentContract
+| DiscountId             | `sdk.Uint`         | How much is the discount
+| Recipient              | `sdk.AccAddress`   | Who is the recipient 
+
+
+```go
+type MsgGrantDiscount struct {
+	SenderDid         did.Did        `json:"sender_did" yaml:"sender_did"`
+	PaymentContractId string         `json:"payment_contract_id" yaml:"payment_contract_id"`
+	DiscountId        sdk.Uint       `json:"discount_id" yaml:"discount_id"`
+	Recipient         sdk.AccAddress `json:"recipient" yaml:"recipient"`
+}
+```
+This message is expected to fail if:
+- any editable field violates the restrictions set for the same field in `MsgGrantDiscount`
+- all editable fields are `"[do-not-modify]"`
+
+## MsgRevokeDiscount 
+
+The owner of a bond can edit some of the bond's parameters using `MsgRevokeDiscount`.
+
+| **Field**              | **Type**           | **Description**                                                                                               |
+|:-----------------------|:-------------------|:--------------------------------------------------------------------------------------------------------------|
+| SenderDid              | `did.Did`          | Who send the transaction 
+| PaymentContractId      | `string`           | ID of the payment_Contract
+| Holder                 | `sdk.AccAddress`   | Address of who's holds the discount
+
+
+
+```go
+type MsgRevokeDiscount struct {
+	SenderDid         did.Did        `json:"sender_did" yaml:"sender_did"`
+	PaymentContractId string         `json:"payment_contract_id" yaml:"payment_contract_id"`
+	Holder            sdk.AccAddress `json:"holder" yaml:"holder"`
+}
+
+```
+This message is expected to fail if:
+- any editable field violates the restrictions set for the same field in `MsgRevokeDiscount`
+- all editable fields are `"[do-not-modify]"`
```

Comment:
This line is from bonds module and does not apply here

Reply:
ok